process/smartContract/scDataGetter.go

```diff
@@ -15,24 +16,33 @@ var maxGasValue = big.NewInt(math.MaxInt64)
 
 // scDataGetter can execute Get functions over SC to fetch stored values
 type scDataGetter struct {
-	vm       vmcommon.VMExecutionHandler
-	mutRunSc sync.Mutex
+	vmContainer process.VirtualMachinesContainer
+	mutRunSc    sync.Mutex
 }
 
 // NewSCDataGetter returns a new instance of scDataGetter
 func NewSCDataGetter(
-	vm vmcommon.VMExecutionHandler,
+	vmContainer process.VirtualMachinesContainer,
 ) (*scDataGetter, error) {
 
-	if vm == nil {
+	if vmContainer == nil || vmContainer.IsInterfaceNil() {
 		return nil, process.ErrNoVM
 	}
 
 	return &scDataGetter{
-		vm: vm,
+		vmContainer: vmContainer,
 	}, nil
 }
 
+func (scdg *scDataGetter) getVMFromAddress(scAddress []byte) (vmcommon.VMExecutionHandler, error) {
+	vmType := scAddress[hooks.NumInitCharactersForScAddress-hooks.VMTypeLen : hooks.NumInitCharactersForScAddress]
```

Comment:
Why not creating an exported func in hooks package that will output the vmType as byte slice? Something like
```GO
vmType := hooks.ComputeVmType(scAddress)
```
and hooks.ComputeVmType will have the code as found on line 38.

Reply:
done